libs/shared/flagd-js-core/src/lib/storage.ts

```diff
@@ -0,0 +1,27 @@
+import { FeatureFlag } from './feature-flag';
+import { parse } from './parser';
+
+/**
+ * The simple contract of the core logics storage layer.
+ */
+export interface Storage {
+  setConfigurations(cfg: string): void;
+
+  getFlag(key: string): FeatureFlag | undefined;
+}
+
+export class StorageImpl implements Storage {
+  private _flags: Map<string, FeatureFlag>;
+
+  constructor() {
+    this._flags = new Map<string, FeatureFlag>();
+  }
+
+  getFlag(key: string): FeatureFlag | undefined {
+    return this._flags.get(key);
+  }
+
+  setConfigurations(cfg: string): void {
```

Comment:
I think we could just as easily make this object immutable, and instantiate a new instance anytime we get a new flagconfig, but I don't have a strong preference here.

Reply:
Given that I have a Java background, my intention here was to reuse the same `FlagdJSCore` instance. but I know things are different in Js world, so I don't have any strong preference. We can simplify and expose set of functions here if we need